### PR TITLE
update eslint-plugin-react version

### PR DESCRIPTION
**Pull request by yashincontrol**
Updated `eslint-plugin-react` and `eslint-plugin-react-hooks` to their latest versions to solve [eslint-plugin-react-hooks](https://github.com/iTwin/eslint-plugin/issues/4)

### CODE REVIEW

File: tools/eslint-plugin/package.json [hunk 1]
```diff
@@ -38,8 +38,8 @@
     "eslint-plugin-jsdoc": "35.1.3",
     "eslint-plugin-jsx-a11y": "6.4.1",
     "eslint-plugin-prefer-arrow": "1.2.3",
-    "eslint-plugin-react": "7.24.0",
-    "eslint-plugin-react-hooks": "4.2.0",
+    "eslint-plugin-react": "^7.32.2",
```
Comment (aruniverse) [inline tools/eslint-plugin/package.json:41]: could you add ^ for the rest of these plugin deps?

@paulius-valiunas any reason these were all pinned?

File: test-apps/display-test-app/package.json [hunk 2]
```diff
@@ -93,8 +93,8 @@
     "eslint-plugin-deprecation": "1.2.1",
     "eslint-plugin-import": "2.23.4",
     "eslint-plugin-prefer-arrow": "1.2.3",
-    "eslint-plugin-react": "7.24.0",
-    "eslint-plugin-react-hooks": "4.2.0",
+    "eslint-plugin-react": "^7.32.2",
```
Comment (aruniverse) [inline test-apps/display-test-app/package.json:96]: honestly should be able to delete all eslint related deps in this test app. should be able to just use the itwin/eslint-plugin itself

**Review by pmconne — APPROVED** [review]

**Comment by aruniverse** [comment]
@mergifyio backport release/3.7.x

File: tools/eslint-plugin/package.json (hunk 1 above)
Comment (yashincontrol) [reply tools/eslint-plugin/package.json:41]: Okay

File: test-apps/display-test-app/package.json (hunk 2 above)
Comment (yashincontrol) [reply test-apps/display-test-app/package.json:96]: Okay, I will try that

**Review by tcobbs-bentley — APPROVED** [review]

File: ui/components-react/src/components-react/tree/controlled/component/TreeNodeRenderer.tsx [hunk 3]
```diff
@@ -70,6 +70,7 @@ export const TreeNodeRenderer = React.memo((props: TreeNodeRendererProps) => {
       nodeEditorRenderer={props.nodeEditorRenderer}
     />
   );
+  TreeNodeRenderer.displayName = "TreeNodeRenderer";
```
Comment (saskliutas) [inline ui/components-react/src/components-react/tree/controlled/component/TreeNodeRenderer.tsx:73]: Can you move this outside of `TreeNodeRenderer`? Now it looks like this is a part of component implementation.

File: tools/eslint-plugin/package.json (hunk 1 above)
Comment (paulius-valiunas) [reply tools/eslint-plugin/package.json:41]: shouldn't we remove this package now that we have https://github.com/iTwin/eslint-plugin/?

Comment (yashincontrol) [reply tools/eslint-plugin/package.json:41]: Yes, but I am not sure when we plan on doing that.

File: ui/components-react/src/components-react/tree/controlled/component/TreeNodeRenderer.tsx (hunk 3 above)
Comment (yashincontrol) [reply ui/components-react/src/components-react/tree/controlled/component/TreeNodeRenderer.tsx:73]: Yes, my bad.

File: tools/eslint-plugin/package.json (hunk 1 above)
Comment (aruniverse) [reply tools/eslint-plugin/package.json:41]: Need to get this into 3.7. I wasn't expecting any new changes to the pkg in 3.x. As soon as this pr completes we will nuke this PKG from the core repo

File: test-apps/display-test-app/package.json (hunk 2 above)
Comment (aruniverse) [reply test-apps/display-test-app/package.json]: hmm you might have to leave eslint in
image
its listed as a peer dep
image

Comment (yashincontrol) [reply test-apps/display-test-app/package.json]: Okay, I'm surprised why `rush lint` didn't show any errors.

**Review by aruniverse — APPROVED** [review]

**Review by ben-polinsky — APPROVED** [review]

**Review by GerardasB — APPROVED** [review]

**Pull request merged**